core/src/main/java/com/pingcap/tikv/hostmap/UriHostMapping.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.hostmap;
+
+import com.pingcap.tikv.util.HostMapping;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UriHostMapping implements HostMapping {
+
+  private static final Logger logger = LoggerFactory.getLogger(UriHostMapping.class.getName());
+  private ConcurrentMap<String, String> hostMapping = null;
```

Comment:
```suggestion
  private final ConcurrentMap<String, String> hostMapping;
```

Reply:
@xuanyu66 Thanks for your review. I will change it.